ipi/engine/motion/geop.py

```diff
@@ -62,249 +157,322 @@ class BFGSMover(object):
+        self.qlist = geop.qlist             
+        self.glist = geop.glist             
+        self.beads = geop.beads
+        self.cell = geop.cell
+        self.forces = geop.forces
+        self.fixcom = geop.fixcom
+        self.fixatoms = geop.fixatoms
 
-        super(GeopMover, self).__init__(fixcom=fixcom, fixatoms=fixatoms)
-
-        # optimization options
-        self.ls_options = ls_options
-        self.tolerances = tolerances
-        self.mode = mode
-        self.max_step = maximum_step
-        self.cg_old_f = cg_old_force
-        self.cg_old_d = cg_old_direction
-        self.invhessian = invhessian
-        self.corrections = corrections
-        self.qlist = qlist
-        self.glist = glist
-
-        self.lm = LineMover()
-        self.bfgsm = BFGSMover()
-
-    def bind(self, ens, beads, nm, cell, bforce, prng):
+        self.lm.bind(self)
+        self.gm.bind(self)
 
-        super(GeopMover, self).bind(ens, beads, nm, cell, bforce, prng)
-        if self.cg_old_f.shape != beads.q.size:
-            if self.cg_old_f.size == 0:
-                self.cg_old_f = np.zeros(beads.q.size, float)
+        if self.old_f.shape != self.beads.q.size:
+            if self.old_f.size == 0:
+                self.old_f = np.zeros(self.beads.q.size, float)
             else:
                 raise ValueError("Conjugate gradient force size does not match system size")
-        if self.cg_old_d.size != beads.q.size:
-            if self.cg_old_d.size == 0:
-                self.cg_old_d = np.zeros(beads.q.size, float)
+        if self.old_d.size != self.beads.q.size:
+            if self.old_d.size == 0:
+                self.old_d = np.zeros(self.beads.q.size, float)
             else:
                 raise ValueError("Conjugate gradient direction size does not match system size")
-        if self.invhessian.size != (beads.q.size * beads.q.size):
+        if self.invhessian.size != (self.beads.q.size * self.beads.q.size):
             if self.invhessian.size == 0:
-                self.invhessian = np.eye(beads.q.size, beads.q.size, 0, float)
+                self.invhessian = np.eye(self.beads.q.size, self.beads.q.size, 0, float)
             else:
                 raise ValueError("Inverse Hessian size does not match system size")
+                
+    def exitstep(self, fx, u0, x):
+        """ Exits the simulation step. Computes time, checks for convergence. """
+        
+        info(" @GEOP: Updating bead positions", verbosity.debug)
+        
+        self.qtime += time.time()
+        
+        # Determine conditions for converged relaxation
+        if (np.absolute((fx - u0) / self.beads.natoms) <= self.tolerances["energy"])\
+                and ((np.amax(np.absolute(self.forces.f)) <= self.tolerances["force"])
+                    or (np.sqrt(np.dot(self.forces.f.flatten() - self.old_f.flatten(),
+                        self.forces.f.flatten() - self.old_f.flatten())) == 0.0))\
```

Comment:
Do I read correctly that the last check (or ... ==0) is the same as saying  "or (self.forces.f.flatten() == self.old_f.flatten())" ? Seems to be an overly complicated way of doing that. Any reasons?